config/archive.go

```diff
@@ -0,0 +1,86 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/contiv/volplugin/errors"
+	"github.com/contiv/volplugin/watch"
+	"github.com/coreos/etcd/client"
+	"golang.org/x/net/context"
+)
+
+func (c *Client) policyArchive(name string) string {
+	return c.prefixed(rootPolicyArchive, name)
+}
+
+func (c *Client) policyArchiveEntry(name string, timestamp string) string {
+	return c.prefixed(rootPolicyArchive, name, timestamp)
+}
+
+// CreatePolicyRevision creates an revision entry in a policy's history.
+func (c *Client) CreatePolicyRevision(name string, policy string) error {
+	timestamp := fmt.Sprint(time.Now().Unix())
+	key := c.policyArchiveEntry(name, timestamp)
+
+	_, err := c.etcdClient.Set(context.Background(), key, policy, &client.SetOptions{PrevExist: client.PrevIgnore})
```

Comment:
Hmm, I guess it technically doesn't matter if a second policy write in the same second doesn't actually generate a revision entry.  Good catch.